clang/include/clang/Basic/BuiltinsAMDGPU.def

```diff
@@ -68,6 +68,7 @@ BUILTIN(__builtin_amdgcn_iglp_opt, "vIi", "n")
 BUILTIN(__builtin_amdgcn_s_dcache_inv, "v", "n")
 BUILTIN(__builtin_amdgcn_buffer_wbinvl1, "v", "n")
 BUILTIN(__builtin_amdgcn_fence, "vUicC*", "n")
+BUILTIN(__builtin_amdgcn_fence_opencl, "vUiUicC*", "n")
```

Comment:
We shouldn't have language specific target builtins. Why isn't the existing fence adequate?

Reply:
I think the motivation here is that we now have an integer "address space mask" argument to the fence. The default should be `0`, to represent "all address spaces", but it can be a smaller subset. This is argument gets translated to the MMRAs. But this shouldn't be OpenCL-specific. We need a scheme that works with all input languages that we care about.